src/test.js

```diff
@@ -25,12 +25,13 @@ import TestReport from "./reports/test";
 export default function Test( settings ) {
 	this.expected = null;
 	this.assertions = [];
-	this.semaphore = 0;
 	this.module = config.currentModule;
 	this.steps = [];
 	this.timeout = undefined;
 	this.data = undefined;
 	this.withData = false;
+	this.asyncNextPauseId = 1;
```

Comment:
Nit: I think "asyncNextPauseId" would read more naturally as "nextAsyncPauseId" or even "asyncPausesCreated" (defaulting to 0 rather than 1).

Reply:
I had mentally namespaced these two properties under an "async" prefix, so it was meant more as "async: nextPauseId". I've dropped the prefix for now as it wasn't adding much.